### ISSUE TITLE
update_management_ssl_cert failed with "The error was: TypeError: warn requires a string not a <class 'NoneType'>"

### ISSUE THREAD

**Issue by iwamitsu** [IBM-Security/isam-ansible-collection#179]
Attempted to udpate the server certificate using update_management_ssl_cert role, but failed due to the following error.

> Error> action does not have the right set of arguments or there is a code bug! Options: isamAppliance=self.isam_server, force=False, certificate="/work/server-cert2.p12"

So updated as follows and tried to run again.

> \- name: Update management ssl certificate Mod
  ibm.isam.isam:
    log: "{{ log_level | default('INFO') }}"
    force: "{{ force | default(False) }}"
    action: ibmsecurity.isam.base.management_ssl_certificate.set
    isamapi:
      certificate: "{{ update_management_ssl_cert_cert }}"
      password: "{{ update_management_ssl_cert_pwd }}"   <-- here
  when: update_management_ssl_cert_cert is defined and update_management_ssl_cert_pwd is defined
  notify: Commit Changes

But it failed with "TypeError: warn requires a string not a <class 'NoneType'>" error.
This role was successful when I ran it a few years ago, has anything changed?

**Comment by tombosmansibm** [comment]
I have a newer version of this , I'll update it in the collection.

**Comment by iwamitsu** [comment]
Is it newer than 1.0.28? 
I think it is simply passing the value of the ibm security api via plugin module, but I couldn't figure out where it was getting caught in the checks.
Could you please share the new module with me?

**Comment by tombosmansibm** [comment]
The new role (base/set_management_ssl_cert) is in the master branch , will be published soon.

**Comment by iwamitsu** [comment]
I tried to execute the new role (base/set_management_ssl_cert), but it failed with "TypeError: warn requires a string not a <class 'NoneType'>" error.

debug(-vvvvv) output:

```The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-local-1pjkhy3he/ansible-tmp-1688443612.092134-5-18031727507669/AnsiballZ_isam.py", line 107, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-local-1pjkhy3he/ansible-tmp-1688443612.092134-5-18031727507669/AnsiballZ_isam.py", line 99, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-local-1pjkhy3he/ansible-tmp-1688443612.092134-5-18031727507669/AnsiballZ_isam.py", line 47, in invoke_module
    runpy.run_module(mod_name='ansible_collections.ibm.isam.plugins.modules.isam', init_globals=dict(_module_fqn='ansible_collections.ibm.isam.plugins.modules.isam', _modlib_path=modlib_path),
  File "/usr/lib64/python3.9/runpy.py", line 225, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib64/python3.9/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible_collections/ibm/isam/plugins/modules/isam.py", line 131, in <module>
  File "/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible_collections/ibm/isam/plugins/modules/isam.py", line 123, in main
  File "/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible/module_utils/basic.py", line 1521, in exit_json
  File "/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible/module_utils/basic.py", line 1491, in _return_formatted
  File "/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible/module_utils/basic.py", line 585, in warn
  File "/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible/module_utils/common/warnings.py", line 18, in warn
    return _formatwarnmsg_impl(msg)
**TypeError: warn requires a string not a <class 'NoneType'>**
fatal: [staff1]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-local-1pjkhy3he/ansible-tmp-1688443612.092134-5-18031727507669/AnsiballZ_isam.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-local-1pjkhy3he/ansible-tmp-1688443612.092134-5-18031727507669/AnsiballZ_isam.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-local-1pjkhy3he/ansible-tmp-1688443612.092134-5-18031727507669/AnsiballZ_isam.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.ibm.isam.plugins.modules.isam', init_globals=dict(_module_fqn='ansible_collections.ibm.isam.plugins.modules.isam', _modlib_path=modlib_path),\n  File \"/usr/lib64/python3.9/runpy.py\", line 225, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.9/runpy.py\", line 97, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib64/python3.9/runpy.py\", line 87, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible_collections/ibm/isam/plugins/modules/isam.py\", line 131, in <module>\n  File \"/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible_collections/ibm/isam/plugins/modules/isam.py\", line 123, in main\n  File \"/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible/module_utils/basic.py\", line 1521, in exit_json\n  File \"/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible/module_utils/basic.py\", line 1491, in _return_formatted\n  File \"/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible/module_utils/basic.py\", line 585, in warn\n  File \"/tmp/ansible_ibm.isam.isam_payload_5e40xkb4/ansible_ibm.isam.isam_payload.zip/ansible/module_utils/common/warnings.py\", line 18, in warn\n    return _formatwarnmsg_impl(msg)\nTypeError: warn requires a string not a <class 'NoneType'>\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}```

I think that this error is not the role problem, but a plugin problem. 
I couldn't figure out where it was getting caught in the checks.

**Comment by tombosmansibm** [comment]
There's indeed a problem in the ibmsecurity code (warnings cannot be None).  I'll fix it there, but in the mean time you could get around the problem by temporarily uninstalling cryptography.
`pip uninstall cryptography`

This disables idempotency, but the role works.

**Comment by tombosmansibm** [comment]
Fixed in ibmsecurity 2023.7.4.0 .

I've noticed that idempotency not always works, but that is a different issue.
